include/veriblock/blockchain/blocktree.hpp

```diff
@@ -148,15 +169,74 @@ struct BlockTree {
       current->height = current->pprev->height + 1;
       current->chainWork = current->pprev->chainWork + getBlockProof(block);
     } else {
-      current->height = 0;
       current->chainWork = getBlockProof(block);
     }
-
     determineBestChain(activeChain_, *current);
 
     return current;
   }
 
+  void invalidateBlockFromChain(Chain<Block>& chain, index_t** block) {
+    if (*block == nullptr) {
+      return;
+    }
+
+    if (!chain.contains(*block) &&
+        chain.tip()->getAncestor((*block)->height) != *block) {
```

Comment:
why do you need this check here?
isn't `contains` enough?

Reply:
no it is not enough, because fork chains dont have all blocks, and it is on that case that we want do remove the old block from the main chain, and as a result we have to remove the parts of the forks.

Reply:
instead of getAncestor use `activeChain.contains(*block)`